truckersmp_cli/utils.py

```diff
@@ -0,0 +1,405 @@
+
+            if (res.status == 301
+                    or res.status == 302
+                    or res.status == 303
+                    or res.status == 307
+                    or res.status == 308):
+                # HTTP redirection
+                u = urllib.parse.urlparse(res.getheader("Location"))
+                if not download_files(
+                  u.netloc, [(u.path, dest, md5), ], (file_count, num_of_files)):
+                    return False
+                # downloaded successfully from redirected URL
+                del files_to_download[0]
+                file_count += 1
+                conn = http.client.HTTPSConnection(host)
+                continue
+            elif res.status != 200:
+                logging.error(
+                  "Server {} responded with status code {}.".format(host, res.status))
+                return False
+
+            lastmod = res.getheader("Last-Modified")
+            content_len = res.getheader("Content-Length")
+
+            with open(dest, "wb") as f:
+                downloaded = 0
+                while True:
+                    buf = res.read(bufsize)
+                    if not buf:
+                        break
+                    downloaded += len(buf)
+                    f.write(buf)
+                    md5hash.update(buf)
+                    if content_len:
+                        progress = "{:,} / {:,}".format(downloaded, int(content_len))
+                    else:
+                        progress = "{:,}".format(downloaded)
+                    print("\r{:40}{:>40}".format(name_getting, progress), end="")
+
+            if md5hash.hexdigest() != md5:
+                print("\r{:40}{:>40}".format(name, "MD5 MISMATCH"))
+                logging.error("MD5 mismatch for {}".format(dest))
+                return False
+
+            # wget-like timestamping for downloaded files
+            if lastmod:
+                timestamp = time.mktime(
+                  time.strptime(lastmod, "%a, %d %b %Y %H:%M:%S GMT")) - time.timezone
+                try:
+                    os.utime(dest, (timestamp, timestamp))
+                except Exception:
+                    pass
+
+            # downloaded successfully
+            print("\r{:40}{:>40}".format(name, "OK"))
+
+            # skip already downloaded files
+            # when trying to download from URL.dlurlalt
+            del files_to_download[0]
+
+            file_count += 1
+    except Exception as e:
+        logging.error("Failed to download https://{}{}: {}".format(host, path, e))
+        return False
+    finally:
+        conn.close()
+
+    return True
+
+
+def get_current_steam_user():
+    """
+    Get the current AccountName with saved login credentials.
+
+    If successful this returns the AccountName of the user with saved credentials.
+    Otherwise this returns None.
+
+    This function depends on the package "vdf".
+    """
+    if not vdf_is_available:
+        return None
+
```

Comment:
Sorry, I missed this:
```suggestion
```
This check was added in this PR. Do you have any reason to do the double check for the splitting?

See also https://github.com/lhark/truckersmp-cli/pull/67#discussion_r429602008

Reply:
No, also forgot about this one. But in retrospect it couldn't hurt to include a check where it actually could fail.